vendor/bat-native-ledger/src/bat/ledger/internal/gemini/gemini.h

```diff
@@ -0,0 +1,111 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_GEMINI_GEMINI_H_
+#define BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_GEMINI_GEMINI_H_
+
+#include <stdint.h>
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "base/containers/flat_map.h"
+#include "base/timer/timer.h"
+#include "bat/ledger/ledger.h"
+
+namespace ledger {
+class LedgerImpl;
+
+namespace endpoint {
+class GeminiServer;
+}
+
+namespace gemini {
+
+struct Transaction {
+  std::string address;
+  double amount;
+  std::string message;
+};
+
+class GeminiTransfer;
+class GeminiAuthorization;
+class GeminiWallet;
+
+using FetchBalanceCallback = std::function<void(type::Result, double)>;
```

Comment:
`#include "bat/ledger/ledger.h"` above has
```
using FetchBalanceCallback =
    std::function<void(type::Result, type::BalancePtr)>;
```
but this class redeclares as
```
using FetchBalanceCallback = std::function<void(type::Result, double)>;
```

Reply:
yea, this is a duplicate in a different namespace and `ledger::FetchBalanceCallback` should be used here

Reply:
`post_balance` in `endpoints/gemini/post_balance/post_balance.ccc` requires a callback `std::function<void(type::Result result, double)>;` so I think `ledger::gemini:FetchBalanceCallback` is used correctly here. Let me know if I'm missing something.

Reply:
Maybe the confusing part is that `ledger::FetchBalanceCallback` appears to return a total balance for all wallets, where `ledger::gemini::FetchBalanceCallback` just returns the Gemini wallet balance.
```
struct Balance {
  double total;
  double user_funds;
  map<string, double> wallets;
};
```

Reply:
@tmancey it looks like this is yet another copy from uphold, etc... so I guess we can group it in with the rest of the duplicate code removal, but also we need to stop using doubles for this stuff. Do we have a ticket for converting to `typedef unsigned _ExtInt(256) uint256_t;` in general?

Reply:
Agreed